scripts/prepare_lambda_deployment.py

```diff
@@ -66,6 +81,101 @@ def patch_cloudwatch_resource(resource_type_name, property_name):
     return input_json
 
 
+def patch_resource_names(tf_config: JSON) -> JSON:
+    """
+    Some Chalice-generated resources have named ending in `-event`. The
+    dash prevents generation of a fully qualified resource name (i.e.,
+    with `config.qualified_resource_name`. This function returns
+    Terraform configuration with names and references updated to
+    remove the trailing `-event`.
+
+    >>> from azul.doctests import assert_json
+    >>> assert_json(patch_resource_names({
+    ...     'resource': {
+    ...         'aws_cloudwatch_event_rule': {
+    ...             'indexercachehealth-event': {
+    ...                'foo': 'abc'
+    ...             },
+    ...             'servicecachehealth-event': {
+    ...                 'bar': '${aws_cloudwatch_event_rule.indexercachehealth-event.foo}'
+    ...             }
+    ...         }
+    ...     }
+    ... }))
+    {
+        "resource": {
+            "aws_cloudwatch_event_rule": {
+                "indexercachehealth": {
+                    "foo": "abc"
+                },
+                "servicecachehealth": {
+                    "bar": "${aws_cloudwatch_event_rule.indexercachehealth.foo}"
+                }
+            }
+        }
+    }
+
+
+    >>> assert_json(patch_resource_names({
+    ...     'resource': {
+    ...         'unaffected_resource_type': {
+    ...             'unaffected_resource_name': {}
+    ...         }
+    ...     }
+    ... }))
+    {
+        "resource": {
+            "unaffected_resource_type": {
+                "unaffected_resource_name": {}
+            }
+        }
+    }
+    """
+    rename = {
+        'aws_cloudwatch_event_rule.indexercachehealth-event': 'indexercachehealth',
+        'aws_cloudwatch_event_target.indexercachehealth-event': 'indexercachehealth',
+        'aws_lambda_permission.indexercachehealth-event': 'indexercachehealth',
+        'aws_cloudwatch_event_rule.servicecachehealth-event': 'servicecachehealth',
+        'aws_cloudwatch_event_target.servicecachehealth-event': 'servicecachehealth',
+        'aws_lambda_permission.servicecachehealth-event': 'servicecachehealth',
+    }
```

Comment:
While it's OK to hard code the list of resources in rename_resources.py, hard-coding it here seems very brittle. There must be a less pedestrian way to code this.

Reply:
My mistake. I thought the purpose of your [fixup](https://github.com/DataBiosphere/azul/pull/1701/commits/63e71d16903a168f06e2c01a651248655b464148#diff-b6226abb933cef28636f31a369c1d7f96b90c211cb9c29f30f3455a95b55ea06R40) was to address this comment.